gridsync/gui/__init__.py

```diff
@@ -1,22 +1,50 @@
 # -*- coding: utf-8 -*-
 
+from typing import List, Tuple
+
+import attr
+
 from gridsync.desktop import notify
 from gridsync.gui.debug import DebugExporter
 from gridsync.gui.main_window import MainWindow
 from gridsync.gui.preferences import PreferencesWindow
 from gridsync.gui.systray import SystemTrayIcon
 from gridsync.gui.welcome import WelcomeDialog
+from gridsync.preferences import Preferences
 
 
+@attr.s
 class Gui:
-    def __init__(self, core):
-        self.core = core
-        self.welcome_dialog = WelcomeDialog(self)
-        self.main_window = MainWindow(self)
-        self.preferences_window = PreferencesWindow()
-        self.systray = SystemTrayIcon(self)
-        self.debug_exporter = DebugExporter(core)
-        self.unread_messages = []
+    core = attr.ib()
+
+    preferences: Preferences = attr.ib(default=attr.Factory(Preferences))
+    unread_messages: List[Tuple] = attr.ib(default=attr.Factory(list))
+
+    welcome_dialog: WelcomeDialog = attr.ib()
+    main_window: MainWindow = attr.ib()
+    preferences_window: PreferencesWindow = attr.ib()
+    systray: SystemTrayIcon = attr.ib()
+    debug_exporter: DebugExporter = attr.ib()
+
+    @welcome_dialog.default
+    def _default_welcome_dialog(self):
```

Comment:
A _very_ small nitpick (since the scope of these wrapped methods is also very small) but consider using a consistent naming scheme here; some of these methods follow the naming pattern of `_x_default` while others follow `_default_x` (I have no preference as to which of the two forms is better -- and won't hold up merging this because of something so trivial -- I just wanted to point it out).

Reply:
Huh.  Yes, I agree consistent naming would be good here.  I was _trying_ to be consistent, in fact, I guess I should have tried harder!